File: sol/solution/0201-0300/0202/Solution.cs
public class Solution {
    public bool IsHappy(int n) {
        int GetNext(int num) {
            int sum = 0;
            while (num > 0) {
                int d = num % 10;
                sum += d * d;
                num /= 10;
            }
            return sum;
        }
        int slow = n, fast = GetNext(n);
        while (fast != 1 && slow != fast) {
            slow = GetNext(slow);
            fast = GetNext(GetNext(fast));
        }
        return fast == 1;
    }
}

File: sol/solution/0201-0300/0202/Solution.rs
impl Solution {
    pub fn is_happy(n: i32) -> bool {
        fn get_next(mut num: i32) -> i32 {
            let mut sum = 0;
            while num > 0 {
                let d = num % 10;
                sum += d * d;
                num /= 10;
            }
            sum
        }
        let (mut slow, mut fast) = (n, get_next(n));
        while fast != 1 && slow != fast {
            slow = get_next(slow);
            fast = get_next(get_next(fast));
        }
        fast == 1
    }
}
